### PR TITLE
Animation

### PR DESCRIPTION
**Pull request by ChrisCruze**
The purpose of this pull request is to improve the animation upon pushing the goal based on feedback from last Sunday's meeting. I essentially promise chained the animations (incl. confetti) with push so that all actions dont fire at the same time. As a result, the animations are more smooth and sequential.

There were a couple animations that I wasnt able to figure out (incl. the goal gliding down the list after completion rather than abruptly disappearing and reappearing)

### CODE REVIEW

File: Components/Atoms/GoalListItem.js [hunk 1]
```diff
@@ -253,21 +254,28 @@ const GoalListItem = ({
...
-          duration: 1500,
-        }).start();
+      pushGoal().then(() => {
+        Animated.timing(moveAnim, {
+          toValue: 0,
+          duration: 0,
+        }).start(() => {
+          Animated.timing(fadeAnim, {
+            toValue: 1,
+            duration: 1500,
```
Comment (anandangalig) [inline Components/Atoms/GoalListItem.js:264]: @ChrisCruze this is like 1.5 seconds right? Can we bring it down to like to 1/3 or 1/2 of a second? Also, I think just the fade animation is good enough. I think we should try it with no moveAnim. 

Animation seems to be slow on Expo/RN. Let's try and minimize it where we can?

File: Components/Molecules/GoalItem.js [hunk 2]
```diff
@@ -93,11 +94,12 @@ const GoalItem = ({
       },
     })
       .then(() => refetch())
-      .catch(e => console.error(e));
-
-    if (totalCount + 1 == cadenceCount) {
-      goalsListConfetti();
-    }
+      .catch(e => console.error(e))
```
Comment (anandangalig) [inline Components/Molecules/GoalItem.js:97]: catch should be very last so it catches all errors that might happen at any of the previous chains, please move it so it runs after all thens are done.

Comment (ChrisCruze) [reply Components/Molecules/GoalItem.js]: Good feedback - I've addressed and updated

**Comment by ChrisCruze** [comment]
Let me know if anything else before merging in

**Pull request merged**